src/collection-facets/facets-template.ts

```diff
@@ -22,34 +26,24 @@ export class FacetsTemplate extends LitElement {
 
   @property({ type: String }) renderOn?: string;
 
-  @property({ type: String }) collectionPagePath: string = '/details/';
-
   @property({ type: Object })
   collectionNameCache?: CollectionNameCacheInterface;
 
-  private facetClicked(e: Event, count: number, negative: boolean) {
-    const target = e.target as HTMLInputElement;
-    const { checked, name, value } = target;
-    if (checked) {
-      this.facetChecked(name as FacetOption, value, count, negative);
+  private facetClicked(e: CustomEvent<FacetEventDetails>) {
+    const { bucket, negative } = e.detail;
+    if (bucket.state === 'none') {
+      this.facetUnchecked(bucket);
     } else {
-      this.facetUnchecked(name as FacetOption, value);
+      this.facetChecked(bucket, negative);
     }
 
-    this.dispatchFacetClickEvent(
-      name as FacetOption,
-      this.getFacetState(checked, negative),
-      negative
-    );
+    this.dispatchFacetClickEvent(e.detail);
   }
 
-  private facetChecked(
-    key: FacetOption,
-    value: string,
```

Comment:
Old code had some inconsistencies around what `key` refers to (the key of the facet group, vs. the key of the bucket). I've tried to normalize our interfaces a bit so that they more consistently indicate which of these is expected.